src/components/accounts/two_factor/TwoFactorVerifyModal.js

```diff
@@ -42,11 +42,10 @@ const TwoFactorVerifyModal = ({ open, onClose }) => {
 
     const handleVerifyCode = async () => {
         const { success, res } = await dispatch(verifyTwoFactor(code))
-        if (success && typeof res !== 'boolean' && res !== false) {
-            onClose(true)
-        } else {
-            onClose(false, new WalletError('Transaction failed with return value: ' + res))
+        if (success === false || res === false) {
```

Comment:
@mattlockyer why is result returned in such a weird way that 2 values needs to be checked? Why not throw error on failure from `verifyTwoFactor`?